lib/Models/Terria.js

```diff
@@ -550,17 +550,25 @@ function generateInitializationUrl(url) {
 
 function loadInitSources(terria, initSources) {
     return when.all(initSources.map(loadInitSource.bind(undefined, terria)), function(initSources) {
-        var promises = [];
+        var promises = [], updateSources = [];
 
         for (var i = 0; i < initSources.length; ++i) {
             var initSource = initSources[i];
             if (!defined(initSource)) {
                 continue;
             }
-            promises.push(terria.addInitSource(initSource));
+
+            if (initSource.catalogOnlyUpdatesExistingItems) {
```

Comment:
Because I need to build a complete map of all the ids in the initial data before I update them with values from the share link.

Without splitting it into two steps like this, generating the map just returns a map of all the root catalog groups as the promises that read the children into those groups haven't executed yet.